torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -991,6 +991,23 @@ class AllocateReuseModifier {
     auto this_tv = alloc_info->alloc_expr->buffer()->as<TensorView>();
     auto reuse_tv = to_reuse->alloc_expr->buffer()->as<TensorView>();
 
+    // Check that either both tv's are vectorized acceses, or neither are.
+    // Vectorized allocations require correct alignment so they can only alias
+    // with other allocations with the right alignment
+    const auto& va = GpuLower::current()->vectorizedAccesses();
+    if ((va.find(this_tv) == va.end()) != (va.find(reuse_tv) == va.end())) {
+      return false;
+    }
+
+    // Shared memory is all aligned to 128 bits, local memory might not be
+    if (this_tv->getMemoryType() == MemoryType::Local &&
```

Comment:
We might want to move this to line 922. This seems to apply for outer sharing as well.

Reply:
@shmsong can you double check I did this right?